kernel/thread.c

```diff
@@ -36,6 +36,38 @@ extern struct _static_thread_data _static_thread_data_list_end[];
 	     thread_data < _static_thread_data_list_end; \
 	     thread_data++)
 
+#if defined(CONFIG_THREAD_MONITOR)
+void k_thread_foreach(thread_user_cb user_cb, void *user_data)
+{
+	struct k_thread *thread;
+
+	__ASSERT(user_cb, "user_cb can not be NULL");
+
+	for (thread = _kernel.threads; thread != NULL;
+			thread = thread->next_thread) {
+		user_cb(thread, user_data);
+	}
+}
+#else
+void k_thread_foreach(thread_user_cb user_cb, void *user_data) { }
+#endif
+
+#if defined(CONFIG_THREAD_STACK_INFO) && defined(CONFIG_THREAD_MONITOR)
+void k_thread_stack_foreach(stack_user_cb user_cb, void *user_data)
+{
+	struct k_thread *thread;
+
+	__ASSERT(user_cb, "user_cb can not be NULL");
+
+	for (thread = _kernel.threads; thread != NULL;
```

Comment:
Same thing for locks here.

Reply:
agree. I will have add the lock while adding as well.